src/game/object/creature.cpp

```diff
@@ -462,13 +475,35 @@ const string &Creature::getWalkAnimation() const {
     }
 }
 
-const string &Creature::getRunAnimation() const {
+string Creature::getRunAnimation() const {
     switch (_modelType) {
         case ModelType::Creature:
             return g_animRunCreature;
         default:
-            return g_animRunCharacter;
+            break;
     }
+
+    // TODO: if (_lowHP) return "runinj" 
+
+    if (_inCombat) {
+        WeaponType type = WeaponType::None;
+        WeaponWield wield = WeaponWield::None;
+        getWeaponInfo(type, wield);
+
+        switch (wield) {
+            case WeaponWield::SingleSaber:
+                return getEquippedItem(kInventorySlotLeftWeapon) ? "runds" : "runss";
```

Comment:
Since return value of `getEquippedItem` is unused, I think we should add and use `bool isSlotEquipped(InventorySlot);` instead.